pkg/provisioner/dependency.go

```diff
@@ -191,3 +86,83 @@ func kind(tpl *bytes.Buffer, env v1alpha1.Environment) error {
 	}
 	return kind.Execute(tpl, env)
 }
+
+// DependencySolver is a struct that holds the dependency graph
+type DependencySolver struct {
+	Dependencies []ProvisionFunc
+	env          v1alpha1.Environment
+}
+
+func NewDependencySolver(env v1alpha1.Environment) *DependencySolver {
+	return &DependencySolver{
+		env: env,
+	}
+}
+
+func (d *DependencySolver) withKubernetes() {
+	switch d.env.Spec.Kubernetes.KubernetesInstaller {
+	case "kubeadm":
+		d.Dependencies = append(d.Dependencies, functions["kubeadm"])
+	case "kind":
+		d.Dependencies = append(d.Dependencies, functions["kind"])
+	case "microk8s":
+		// reset the graph to only include microk8s
+		d.Dependencies = nil
```

Comment:
Does using `microk8s` not also depend on the container toolkit?

Reply:
no, is a canonical snap pkg. it simply needs a fresh ubuntu system, and it will do everything